app/user.go

```diff
@@ -1691,9 +1691,28 @@ func (a *App) GetUserAnalytics(userId string, teamId string, name string) (*mode
 		// Return and exit from GetUserAnalytics with error
 		return nil, userLearningGroups.Err
 	} else {
+
+		userLearningGroupsData := userLearningGroups.Data.([]*model.LearningGroup)
+
+		// Create LearningGroupMember structs for each member in each learning group
+		// Loop over learning groups, and for each, loop over its members
+		// Note: members is a double delimited string of users (; delimited) and each user's id
+		// 			 and username (, delimited)
+		for _, userLearningGroup := range userLearningGroupsData {
+			usersInLearningGroup := strings.Split(*userLearningGroup.MembersDelimitedString, ";")
+			membersArray := make([]*model.LearningGroupMember, len(usersInLearningGroup))
+
+			for j, user := range usersInLearningGroup {
+				userData := strings.Split(user, ",")
+				membersArray[j] = &model.LearningGroupMember{Id: userData[0], Username: userData[1]}
```

Comment:
Here, I don't see a great reason to use an array of pointers to LearningGroupMember structs rather than just using an array of LearningGroupMember structs? Maybe this makes JSON serialization easier later on? Otherwise, seems like the pointers aren't getting us anything except fragmentation and extra dereferencing?

Reply:
I guess I just opted for pointers whenever I was passing data around. My limited understanding of them, is that is can be better to pass a reference than the value.

Reply:
That's not an unreasonable default. There's a tradeoff though. Two basic issues:

-> Using pointers means, all else being equal, two operations to get at data that would otherwise have cost one operation to access.
-> Allocating an array of pointers, and then making each point to an allocated object _can_ cause lots of small heap allocations. Heap allocation has a high overhead, so this will be much slower than the 1 big heap allocation caused to allocate an array of structs. 

In languages like C or C++ it's pretty easy to figure out when these tradeoffs are worth it. In Go, there's an extra complication, because the language specification makes no promises about when heap allocation will occur. Instead, it's going to perform "escape analysis", looking to see whether pointers to an allocated object "escape" the local context in which the object is created. In this case, they will (the array escapes, along with all the pointers in it), so I _think_ it'll do heap allocation of all the structs. My knowledge of this memory model is still far from perfect though...

Anyway, it's probably not a huge deal if it's a pain to change, but I _think_ this is going to be fairly expensive (like ~100x slower than it would otherwise be).

Reply:
I also think this doesn't need to be an array of pointers.

One possible case where an array of pointers would be better than an array of the objects (structs) is when you were re-ordering or inserting/deleting often into the array. That doesn't happen with this array (I don't think).

Reply:
Oh, sorry, the main reason that I am in this habit is:
```
// An array of pointers to LearningGroupMember structs for every member of this learning group (channel)
	Members []*LearningGroupMember `json:"members"`
```
In the model, having the pointers allows it to be nil, which might be the case. I think the omit empty works for this as well. Maybe I will try that.

Reply:
So I think handling null values with pointers is only really necessary for sql results. I've changed this to not use pointers.